pkg/cloudprovider/servicecontroller/servicecontroller.go

```diff
@@ -445,13 +464,39 @@ func portsEqual(x, y *api.Service) bool {
 	if err != nil {
 		return false
 	}
-	if len(xPorts) != len(yPorts) {
+	return intSlicesEqual(xPorts, yPorts)
+}
+
+func intSlicesEqual(x, y []int) bool {
```

Comment:
It could be reusable, but would be a bit of a trap since it can sometimes modify the inputs if they have the same length and aren't already sorted. I'm hesitant to move it somewhere common for that reason.